autotest.pm

```diff
@@ -372,6 +372,7 @@ sub runalltests {
         eval { $t->runtest; };
         my $error = $@;    # save $@, it might be overwritten
         $t->save_test_result();
+        my $next_test = $testindex eq $#testorder ? undef : $testorder[$testindex + 1];
```

Comment:
Why `eq` and not `==`?

Reply:
Just `my $next_test = $testorder[$testindex + 1]` would have worked, too. Why the extra check? It's perl, not python ;-)

Reply:
Ok, sure. @kalikiana and me were just too cautious ;) Created https://github.com/os-autoinst/os-autoinst/pull/1989